src/wwElement.vue

```diff
@@ -72,8 +73,14 @@ export default {
             type: 'string',
             defaultValue: computed(() => String(props.content.selectedDate || '')),
         });
+        const { value: currentViewObj, setValue: setCurrentViewObj } = wwLib.wwVariable.useComponentVariable({
+            uid: props.uid,
+            name: 'currentView',
+            type: 'object',
+            defaultValue: {},
+        });
 
-        return { selectedDate, setSelectedDate };
+        return { selectedDate, setSelectedDate, currentViewObj, setCurrentViewObj };
     },
     data: () => ({
         currentView: null,

```

Comment:
Peut etre refacto pour supprimer le currentView local, il ne sert plus a rien, tu pourras renommer ton currentViewObj en currentView, et retirer plus bas le this.currentView =

Reply:
Effectivement tu as raison, je fais le fix

Reply:
@AlexisJC Je suis allé un peu vite à répondre, en fait currentViewObj est l'objet complete de la vue (type, start date etc) et currentView c'est simplement le type.

Peut-être trouverais-tu ça plus clair de mieux distinguer les noms ?

Exemple pour la propriété defaultView, on ne choisi que le type, pas l'objet entier.